lib/interface_quda.cpp

```diff
@@ -5068,347 +4773,214 @@ void computeCloverForceQuda(void *h_mom, double dt, void **h_x, void **, double
   if (inv_param->use_resident_solution) solutionResident.clear();
 #endif
   delete dirac;
-  profileCloverForce.TPSTOP(QUDA_PROFILE_FREE);
 
-  profileCloverForce.TPSTOP(QUDA_PROFILE_TOTAL);
+  popProfile();
 }
 
-void updateGaugeFieldQuda(void* gauge,
-			  void* momentum,
-			  double dt,
-			  int conj_mom,
-			  int exact,
-			  QudaGaugeParam* param)
+void updateGaugeFieldQuda(void* gauge, void* momentum, double dt, int conj_mom, int exact, QudaGaugeParam* param)
 {
-  profileGaugeUpdate.TPSTART(QUDA_PROFILE_TOTAL);
-
+  pushProfile(profileGaugeUpdate);
   checkGaugeParam(param);
 
-  profileGaugeUpdate.TPSTART(QUDA_PROFILE_INIT);
-
   // create the host fields
   GaugeFieldParam gParam(*param, gauge, QUDA_SU3_LINKS);
   gParam.location = QUDA_CPU_FIELD_LOCATION;
-  gParam.site_offset = param->gauge_offset;
-  gParam.site_size = param->site_size;
   bool need_cpu = !param->use_resident_gauge || param->return_result_gauge;
-  cpuGaugeField *cpuGauge = need_cpu ? new cpuGaugeField(gParam) : nullptr;
+  GaugeField cpuGauge = need_cpu ? GaugeField(gParam) : GaugeField();
 
-  GaugeFieldParam gParamMom(*param, momentum);
-  gParamMom.reconstruct = (gParamMom.order == QUDA_TIFR_GAUGE_ORDER || gParamMom.order == QUDA_TIFR_PADDED_GAUGE_ORDER) ?
```

Comment:
Similar comment as above, I assume this logic has been shifted elsewhere/internally?

Reply:
Yep.